sdk/hybridnetwork/Azure.ResourceManager.HybridNetwork/src/Generated/Models/NetworkFunctionPropertiesFormat.cs

```diff
@@ -0,0 +1,88 @@
+        {
+            ProvisioningState = provisioningState;
+            PublisherName = publisherName;
+            PublisherScope = publisherScope;
+            NetworkFunctionDefinitionGroupName = networkFunctionDefinitionGroupName;
+            NetworkFunctionDefinitionVersion = networkFunctionDefinitionVersion;
+            NetworkFunctionDefinitionOfferingLocation = networkFunctionDefinitionOfferingLocation;
+            NetworkFunctionDefinitionVersionResourceReference = networkFunctionDefinitionVersionResourceReference;
+            NfviType = nfviType;
+            NfviId = nfviId;
+            AllowSoftwareUpdate = allowSoftwareUpdate;
+            ConfigurationType = configurationType;
+            RoleOverrideValues = roleOverrideValues;
+        }
+
+        /// <summary> The provisioning state of the network function resource. </summary>
+        public ProvisioningState? ProvisioningState { get; }
+        /// <summary> The publisher name for the network function. </summary>
+        public string PublisherName { get; }
+        /// <summary> The scope of the publisher. </summary>
+        public PublisherScope? PublisherScope { get; }
+        /// <summary> The network function definition group name for the network function. </summary>
+        public string NetworkFunctionDefinitionGroupName { get; }
+        /// <summary> The network function definition version for the network function. </summary>
+        public string NetworkFunctionDefinitionVersion { get; }
+        /// <summary> The location of the network function definition offering. </summary>
+        public string NetworkFunctionDefinitionOfferingLocation { get; }
+        /// <summary>
+        /// The network function definition version resource reference.
+        /// Please note <see cref="DeploymentResourceIdReference"/> is the base class. According to the scenario, a derived class of the base class might need to be assigned here, or this property needs to be casted to one of the possible derived classes.
+        /// The available derived classes include <see cref="OpenDeploymentResourceReference"/> and <see cref="SecretDeploymentResourceReference"/>.
+        /// </summary>
+        public DeploymentResourceIdReference NetworkFunctionDefinitionVersionResourceReference { get; set; }
+        /// <summary> The nfvi type for the network function. </summary>
+        public NfviType? NfviType { get; set; }
+        /// <summary> The nfviId for the network function. </summary>
+        public string NfviId { get; set; }
```

Comment:
What's the format of this property value?
- If it's a GUID, use type [Guid](https://learn.microsoft.com/en-us/dotnet/api/system.guid?view=net-6.0). See example [here](https://github.com/Azure/azure-sdk-for-net/blob/bd953fc7a97bf617c892ce6314b50126b4021a4f/sdk/maps/Azure.ResourceManager.Maps/src/autorest.md?plain=1#L25).
- If it's a resource identifier, use type [ResourceIdentifier](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/ResourceIdentifier.cs). See example [here](https://github.com/Azure/azure-sdk-for-net/blob/bd953fc7a97bf617c892ce6314b50126b4021a4f/sdk/search/Azure.Search.Documents/src/autorest.md?plain=1#L106-L115).